src/html.ml

```diff
@@ -1,5 +1,29 @@
 open Ast
 
+type printer =
+  {
+    start: printer          -> Buffer.t -> inline block list                                             -> unit;
+    finish: printer         -> Buffer.t                                                                  -> unit;
```

Comment:
This one does not need to be part of the API. One can always override `document` (i.e. `start` above) to perform some work after the processing of the document is finish.